src/tests/selectionpositionresolver_test.cpp

```diff
@@ -0,0 +1,161 @@
+    : QObject(parent)
+    , m_doc(std::make_unique<TextEditor::TextDocument>())
+{}
+
+QTextCursor SelectionPositionResolverTests::getInitializedCursor(const QByteArray &content,
+                                                                 const int position)
+{
+    m_doc->setContents(content);
+    m_doc->setFilePath(Utils::FileName::fromString("myPath"));
+
+    QTextCursor cursor(m_doc->document());
+    cursor.setPosition(position);
+
+    return cursor;
+}
+
+void SelectionPositionResolverTests::test_emptyDocument()
+{
+    auto cursor = getInitializedCursor("", 0);
+
+    m_resolver.resolve(cursor);
+    QCOMPARE(m_resolver.startPosition(), 0);
+    QCOMPARE(m_resolver.endPosition(), 0);
+}
+
+void SelectionPositionResolverTests::test_singleWord()
+{
+    auto cursor = getInitializedCursor("AAAA BBBB CCCC", 6);
+
+    m_resolver.resolve(cursor);
+    QCOMPARE(m_resolver.startPosition(), 5);
+    QCOMPARE(m_resolver.endPosition(), 9);
+}
+
+void SelectionPositionResolverTests::test_singleWordOnlyHyphens()
+{
+    auto cursor = getInitializedCursor("---- ---- ----", 6);
+
+    m_resolver.resolve(cursor);
+    QCOMPARE(m_resolver.startPosition(), 5);
+    QCOMPARE(m_resolver.endPosition(), 9);
+}
+
+void SelectionPositionResolverTests::test_wordWithHyphenInMiddle()
+{
+    auto cursor = getInitializedCursor("AAAA-BBBB-CCCC", 6);
+
+    m_resolver.resolve(cursor);
+    QCOMPARE(m_resolver.startPosition(), 0);
+    QCOMPARE(m_resolver.endPosition(), 14);
+}
+
+void SelectionPositionResolverTests::test_wordWithMultipleHyhpens()
+{
+    auto cursor = getInitializedCursor("AAAA-----BBBB-----CCCC", 1);
+
+    m_resolver.resolve(cursor);
+    QCOMPARE(m_resolver.startPosition(), 0);
+    QCOMPARE(m_resolver.endPosition(), 22);
+}
+
+void SelectionPositionResolverTests::test_selectHyphensAsPartOfWord()
+{
+    auto cursor = getInitializedCursor("AAAA-----BBBB-----CCCC", 6);
+
+    m_resolver.resolve(cursor);
+    QCOMPARE(m_resolver.startPosition(), 0);
+    QCOMPARE(m_resolver.endPosition(), 22);
+}
+
+void SelectionPositionResolverTests::test_wordWithHyphensAtEnd()
+{
+    auto cursor = getInitializedCursor("AAAA---------", 1);
+
+    m_resolver.resolve(cursor);
+    QCOMPARE(m_resolver.startPosition(), 0);
+    QCOMPARE(m_resolver.endPosition(), 13);
+}
+
+void SelectionPositionResolverTests::test_selectHyphensAsWordEnding()
+{
+    auto cursor = getInitializedCursor("AAAA---------", 6);
+
+    m_resolver.resolve(cursor);
+    QCOMPARE(m_resolver.startPosition(), 0);
+    QCOMPARE(m_resolver.endPosition(), 13);
+}
+
+void SelectionPositionResolverTests::test_wordWithHyphensAtStart()
+{
+    auto cursor = getInitializedCursor("---------AAAA", 12);
+
+    m_resolver.resolve(cursor);
+    QCOMPARE(m_resolver.startPosition(), 0);
+    QCOMPARE(m_resolver.endPosition(), 13);
+}
+
+void SelectionPositionResolverTests::test_selectHyphensAsWordStarting()
+{
+    auto cursor = getInitializedCursor("---------AAAA", 2);
```

Comment:
hmm `--` is a comment start in ASN.1/ACN, so maybe we should stop at single `-`?